tool/tctl/common/auth_command.go

```diff
@@ -505,11 +510,66 @@ func (a *AuthCommand) GenerateCRLForCA(ctx context.Context, clusterAPI authComma
 		return trace.Wrap(err)
 	}
 
-	crl, err := clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
+	authorities, err := clusterAPI.GetCertAuthorities(ctx, certType, false)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
+	if a.output == "" {
+		l := len(authorities)
+		switch {
+		case l > 1:
+			return trace.BadParameter("found %d authorities, use --out to export all CRLs", l)
+		case l == 1 && len(authorities[0].GetActiveKeys().TLS) > 1:
+			return trace.BadParameter("CA has multiple active keys, use --out to export all CRLs")
+		}
+	}
+
+	if a.output != "" {
+		// collect the CRLs ahead of time so we can print a message
+		// like we do with tctl auth export
+		type output struct{ cert, crl []byte }
+		var results []output
+		for _, authority := range authorities {
+			for i, keypair := range authority.GetActiveKeys().TLS {
+				if len(keypair.CRL) == 0 {
+					fmt.Fprintf(os.Stderr, "keypair %v is missing CRL for %v authority %v, generating legacy fallback", i, authority.GetType(), authority.GetName())
+					crl, err := clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
+					if err != nil {
+						return trace.Wrap(err)
+					}
+					results = append(results, output{keypair.Cert, crl})
```

Comment:
IIUC if there are multiple TLS keys with no CRL this will still write them to the file name with the key ID and there will be multiple probably identical CRLs, is that the behaviour we want?